.github/workflows/backend_tests.yml

```diff
@@ -27,31 +27,19 @@ jobs:
         with:
           python-version: '3.8.15'
           architecture: 'x64'
-      - name: Cache node modules and third_party/static
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: |
-            /home/runner/work/oppia/yarn_cache
-            /home/runner/work/oppia/oppia/third_party/static
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock', 'dependencies.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - name: Install Oppia dependencies
-        uses: ./.github/actions/install-oppia-dependencies
-        if: startsWith(github.head_ref, 'update-changelog-for-release') == false
-      - name: Install packages for optimized coverage
-        if: startsWith(github.head_ref, 'update-changelog-for-release') == false
-        run: sudo apt-get install -y python-dev-is-python3 gcc
-      - name: Install coverage and configparser
+      - name: Free disk space
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY" 
+      - name: Initialze Containers
         if: startsWith(github.head_ref, 'update-changelog-for-release') == false
-        run: pip install coverage configparser
+        run: |
+          make build
+          sudo make run-devserver
+          make stop
```

Comment:
Can we make these into actions that we reuse, like we did with `.github/actions/install-oppia-dependencies`? That would help reduce code duplication in the workflows